bluemira/equilibria/find.py

```diff
@@ -334,80 +294,82 @@ def find_OX_points(x, z, psi, limiter=None, x_min=None):  # noqa :N802
     """
     d_x, d_z = x[1, 0] - x[0, 0], z[0, 1] - z[0, 0]  # Grid resolution
     x_m, z_m = (x[0, 0] + x[-1, 0]) / 2, (z[0, 0] + z[0, -1]) / 2  # Grid centre
+    nx, nz = psi.shape  # Grid shape
+    radius = min(0.5, 2 * np.hypot(d_x, d_z))  # Search radius
 
-    if x_min is None:
-        f = RectBivariateSpline(x[:, 0], z[0, :], psi)  # Spline for psi interpolation
-    else:
-        # Truncate grid to avoid many OX points on solenoid (CREATE relic)
-        i_x = np.argmin(abs(x[:, 0] - x_min)) - 1
-        f = RectBivariateSpline(x[i_x:, 0], z[0, :], psi[i_x:, :])
-        x = x[i_x:0]
-        psi = psi[i_x:, :]
+    # Splines for interpolation
+    f_psi = RectBivariateSpline(x[:, 0], z[0, :], psi)
 
-    nx, nz = psi.shape  # Grid shape (including truncation)
+    def f_Bx(x, z):
+        return -f_psi(x, z, dy=1, grid=False) / x
 
-    radius = min(0.5, 2 * (d_x ** 2 + d_z ** 2))  # Search radius
+    def f_Bz(x, z):
+        return f_psi(x, z, dx=1, grid=False) / x
 
-    def f_bp(x_opt):
-        """
-        Poloidal field optimiser function handle
-        \t:math:`B_{p} = B_{x}^{2}+B_{z}^{2}`
+    def f_Bp(x, z):
+        return np.hypot(f_Bx(x, z), f_Bz(x, z))
 
-        Notes
-        -----
-        No points square-rooting here, we don't care about actual values.
-        """
-        return (-f(x_opt[0], x_opt[1], dy=1, grid=False) / x_opt[0]) ** 2 + (
-            f(x_opt[0], x_opt[1], dx=1, grid=False) / x_opt[0]
-        ) ** 2
+    Bp2 = f_Bx(x, z) ** 2 + f_Bz(x, z) ** 2
 
-    Bp2 = f_bp(np.array([x, z]))
+    i_local, j_local = find_local_minima(Bp2)
 
     points = []
-    for i, j in zip(*find_local_minima(Bp2)):
+    for i, j in zip(i_local, j_local):
         if i > nx - 3 or i < 3 or j > nz - 3 or j < 3:
             continue  # Edge points uninteresting and mess up S calculation.
 
-        if nx * nz <= 4225:  # scipy method faster on small grids
-            point = find_local_Bp_minima_scipy(f_bp, x[i, j], z[i, j], radius)
+        if f_Bp(x[i, j], z[i, j]) > 1.0:  # T
+            continue
 
-        else:  # Local Newton/Powell CG method faster on large grids
-            point = find_local_Bp_minima_cg(f, x[i, j], z[i, j], radius)
+        point = find_local_Bp_minima_cg(f_psi, x[i, j], z[i, j], radius)
 
         if point:
             points.append(point)
 
     points = drop_space_duplicates(points)
 
-    o_points, x_points = triage_OX_points(f, points)
+    o_points, x_points = triage_OX_points(f_psi, points)
 
     if len(o_points) == 0:
         print("")  # stdout flusher
```

Comment:
All our logging goes to `stderr` at the moment so these are a bit redundant

Reply:
in fact it probably leads in to needing a classified logging system so we can track things like this. Ignore for now